src/lib.rs

```diff
@@ -57,12 +45,8 @@ pub fn init_graph(graph_bytes: &[u8]) -> eyre::Result<Graph> {
     })
 }
 
-/// Calculate witness based on serialized graph and inputs
-pub fn calculate_witness(
-    input_list: HashMap<String, Vec<U256>>,
-    graph: &Graph,
-) -> eyre::Result<Vec<U256>> {
-    // Calculate number of inputs from graph
+/// Calculates the number of needed inputs and allocates vec
+pub fn generate_inputs_buffer(graph: &Graph) -> Vec<U256> {
```

Comment:
Jolt (and likely other consumers using this pattern), will parallelize calls to witness::populate_inputs, graph::evaluate).

As a result maybe a better pattern is:
```
let buffer_size = witness::get_buffer_size(&graph); // Does the max work of `generate_inputs_buffer()`

let wtns = inputs.par_iter().map(|input| {
    let buffer = witness::generate_input_buffer(&list, &mapping);
    graph::evaluate(&graph.nodes, buffer, &graph.signals)
}).collect();
```

It does require reallocation for the `Vec<U256>` buffer, but I think the parallelism is worth it. (Otherwise the threads have to share the buffer).